lms/djangoapps/discussion/rest_api/tests/test_api.py

```diff
@@ -3565,3 +3575,171 @@ def test_group_access(self, role_name, course_is_cohorted, thread_group_state):
+                parent_location=self.sequential.location,
+                category='vertical',
+                display_name=f'vertical-{idx}',
+                start=datetime(2015, 4, 1, tzinfo=UTC),
+            )
+            for idx in range(10)
+        ]
+        staff_only_unit = ItemFactory.create(
+            parent_location=self.sequential.location,
+            category='vertical',
+            display_name='staff-vertical-1',
+            metadata=dict(visible_to_staff_only=True),
+        )
+        self.course_key = course_key = self.course.id
+        self.config = DiscussionsConfiguration.objects.create(context_key=course_key, provider_type=Provider.OPEN_EDX)
+        topic_links = []
+        update_discussions_settings_from_course_task(str(self.course_key))
+        self.staff_only_id = DiscussionTopicLink.objects.filter(
+            usage_key__in=[staff_only_unit.location]
+        ).values_list(
+            'external_id', flat=True,
+        ).get()
+        topic_id_query = DiscussionTopicLink.objects.filter(context_key=course_key).values_list(
+            'external_id', flat=True,
+        )
+        topic_ids = list(topic_id_query.order_by('ordering'))
+        topic_ids.remove(self.staff_only_id)
+        topic_ids_by_name = list(topic_id_query.order_by('title'))
+        topic_ids_by_name.remove(self.staff_only_id)
+        deleted_topic_ids = [f'disabled-topic-{idx}' for idx in range(10)]
+        for idx, topic_id in enumerate(deleted_topic_ids):
+            usage_key = course_key.make_usage_key('vertical', topic_id)
+            topic_links.append(
+                DiscussionTopicLink(
+                    context_key=course_key,
+                    usage_key=usage_key,
+                    title=f"Discussion on {topic_id}",
+                    external_id=topic_id,
+                    provider_id=Provider.OPEN_EDX,
+                    ordering=idx,
+                    enabled_in_context=False
+                )
+            )
+        DiscussionTopicLink.objects.bulk_create(topic_links)
+        self.topic_ids = topic_ids
+        self.topic_ids_by_name = topic_ids_by_name
+        self.user = UserFactory.create()
+        self.staff = AdminFactory.create()
+        self.topic_stats = {
+            topic_id: dict(discussion=random.randint(0, 10), question=random.randint(0, 10))
+            for topic_id in self.topic_ids
+        }
+        patcher = mock.patch(
+            'lms.djangoapps.discussion.rest_api.api.get_course_commentable_counts',
+            mock.Mock(return_value=self.topic_stats),
+        )
+        patcher.start()
+        self.addCleanup(patcher.stop)
+
+    def test_default_response(self):
+        """
+        Test that the standard response contains the correct number of items
+        """
+        topics_list = get_course_topics_v2(course_key=self.course_key, user=self.user)
+        assert len(topics_list) == len(self.topic_ids)
+
+    def test_staff_response(self):
```

Comment:
[inform] I just removed this test because it is flaky. I do not know if that means the functionality is flaky, or just the test. Unfortunately, I also merged the test removal without a PR by mistake. Here is the commit with the removal: https://github.com/openedx/edx-platform/commit/6c7618fed0ec43995146047cc44097bc75720dc9